api/analyzedphenotypes.datatype.api.inc

```diff
@@ -0,0 +1,286 @@
+    'colour'         => array('name' => 'color',     'type' => 'text'),
+    'percent'         => array('name' => 'percentage',  'type' => 'number'),
+    'comments'         => array('name' => 'comments',     'type' => 'text'),
+    'scale 1-5'         => array('name' => 'scale 1-5',     'type' => 'scale_1-5'),
+    'scale 1-10'         => array('name' => 'scale 1-10',     'type' => 'scale_1-10'),
+    'average count'       => array('name' => 'average count',   'type' => 'number'),
+
+    // Define unit here...
+  );
+
+  return $datatype;
+}
+
+
+///////
+
+
+/**
+ * CONTROL/SWITCHBOARD.
+ * Pass a request to a corresponding function.
+ * @TODO: leave out when api reorganize/update is complete and use functions below.
+ */
+function analyzedphenotypes_datatypeprop($property = null, $parameter = null) {
+  // Process request.
+  $command = array(
+    'unit_type' => array(
+      'func' => 'ap_define_datatypes',
+      'unit' => null,
+      'val'  => null,
+    ),
+    'alphanumeric' => array(
+      'func' => 'ap_validate_unitvalue',
+      'unit' => $parameter['unit'],
+      'val'  => $parameter['value'],
+    ),
+    'trait' => array(
+      'func' => 'ap_validate_unitvalue',
+      'unit' => $parameter['unit'],
+      'val'  => $parameter['value'],
+    ),
+    'text' => array(
+      'func' => 'ap_validate_unitvalue',
+      'unit' => $parameter['unit'],
+      'val'  => $parameter['value'],
+    ),
+    'number' => array(
+      'func' => 'ap_validate_unitvalue',
+      'unit' => $parameter['unit'],
+      'val'  => $parameter['value'],
+    ),
+    'four_digit_year' => array(
+      'func' => 'ap_validate_unitvalue',
+      'unit' => $parameter['unit'],
+      'val'  => $parameter['value'],
+    ),
+    'number_no_zero' => array(
+      'func' => 'ap_validate_unitvalue',
+      'unit' => $parameter['unit'],
+      'val'  => $parameter['value'],
+    ),
+    'scale_1-5' => array(
+      'func' => 'ap_validate_unitvalue',
+      'unit' => $parameter['unit'],
+      'val'  => $parameter['value'],
+    ),
+    'scale_1-10' => array(
+      'func' => 'ap_validate_unitvalue',
+      'unit' => $parameter['unit'],
+      'val'  => $parameter['value'],
+    ),
+    'y/n/?' => array(
+      'func' => 'ap_validate_unitvalue',
+      'unit' => $parameter['unit'],
+      'val'  => $parameter['value'],
+    ),
+    'date' => array(
+      'func' => 'ap_validate_unitvalue',
+      'unit' => $parameter['unit'],
+      'val'  => $parameter['value'],
+    ),
+  );
+
+  return call_user_func($command[$property]['func'], $command[$property]['unit'], $command[$property]['val']);
+}
+
+
+///////
+
+
+/**
+ * FUNCTION:
+ * Fetch the data type of a unit. The data type will be used to validate value/data to ensure
+ * that data matches the unit the trait is measuring.
+ * Dependencies: ap_define_datatype().
+ *
+ * @param $unit
+ *   A string value, the unit of a trait.
+ *
+ * @return
+ *   String, data type of unit.
```

Comment:
I think the return value is actually an array with 'name' and 'type' defined. If you want this to actually return the string type it should be `$datatype[ $unit ]['type']` below.

Reply:
👍